### PR TITLE
[IMP] stock_location_orderpoint: Add domain to filter locations

### PR DESCRIPTION
**Pull request by rousseldenis**
Depends on:

- https://github.com/OCA/stock-logistics-availability/pull/15

### CODE REVIEW

**Comment by rousseldenis** [comment]
@jbaudoux

File: stock_location_orderpoint/__manifest__.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
         "demo/stock_location_orderpoint.xml",
     ],
     "depends": [
+        "stock_available_base_exclude_location",
```
Comment (rousseldenis) [inline stock_location_orderpoint/__manifest__.py:27]: @jbaudoux As this adds a new dependency, I think a new module to extend is a good idea.

File: stock_location_orderpoint/tests/test_sublocation_orderpoint.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from .common import TestLocationOrderpointCommon
+
+
+class TestSubLocationOrderpoint(TestLocationOrderpointCommon):
```
Comment (lmignon) [inline stock_location_orderpoint/tests/test_sublocation_orderpoint.py:6]: @rousseldenis This file is not imported into __init__ and tests are therefore not executed by CI Moreover, the tests are failing.

**Comment by lmignon** [comment]
@rousseldenis I rebased your PR on master and fixed some performance issues in #9...

**Comment by rousseldenis** [comment]
See #9

**Pull request closed without merging**